backend/src/Designer/Services/Implementation/TextsService.cs

```diff
@@ -327,18 +347,83 @@ private async Task UpdateKeysInLayoutsInLayoutSet(string org, string app, string
                 }
                 foreach (var layoutObject in layoutArray)
                 {
-                    foreach (TextIdMutation mutation in keyMutations.Where(_ => layoutObject["textResourceBindings"] is not null))
+                    foreach (TextIdMutation mutation in keyMutations)
                     {
-                        layoutObject["textResourceBindings"] = UpdateKey(layoutObject["textResourceBindings"], mutation);
+                        UpdateKeyInLayoutObject(layoutObject, mutation);
                     }
+                    await altinnAppGitRepository.SaveLayout(layoutSetName, layoutName, layout);
```

Comment:
It looks like this code is executed for every layout and component. I suggest optimizing it by updating only the layouts impacted by the changes and updating them only once

Reply:
Good catch and suggestion! This call was supposed to be one layer up, saving only once per layout. I fixed this, and also implemented your suggestion to save only if changes have been made 👍

Reply:
I’m wondering if doing a deep clone and deep comparison could become slow/resource-intensive with large layouts, especially since it’s done here for every layout in the layout set. 🤔 
Could it be possible to do something like this instead:

```
bool updated = false;
foreach (JsonNode layoutObject in layoutArray)
{
    foreach (TextIdMutation mutation in keyMutations)
    {
        updated |= UpdateKeyInLayoutObject(layoutObject, mutation);
    }
}
if (updated)
{
    await altinnAppGitRepository.SaveLayout(layoutSetName, layoutName, layout);
}
```

Reply:
I was hesitant to do something like this initially, as i thought it would lead to harder to read code, but i don't think it turned out too bad. Updated the code, let me know what you think 👍